services/galley/src/Galley/API/Update.hs

```diff
@@ -1126,7 +1126,7 @@ replaceMembers lusr zcon qcnv (InviteQualified invitedUsers role) = do
       invitedMembersSet = Set.fromList $ toList invitedUsers
       ugMembers = concatMap (fmap (flip Qualified (tDomain lusr)) . V.toList . runIdentity . (.members)) (V.toList ugs)
       -- the invited users plus all user group members should stay
-      allUsersThatShouldStay = Set.fromList $ toList $ appendList invitedUsers ugMembers
+      allUsersThatShouldStay = Set.fromList $ invitedUsers <> ugMembers
```

Comment:
This seems cleaner, but I don't see how this is a different behavior, and how that fixes the problem...